### PR TITLE
[issue-21] Intégration de Connect et SLN

### PR DESCRIPTION
**Pull request by Jean-Beru**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ web/bundles/
 log/*
 web/css/*
 web/js/*
+/nbproject/private/
```
Comment (fgueguen) [inline .gitignore]: missing newline at end of file

Comment (mRoca) [inline .gitignore]: https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

Comment (mRoca) [inline .gitignore]: Think to add a newline at the files' end => use PhpStorm or run php-cs-fixer ;)

File: app/config/security.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ security:
             connect_memory:
                 users:
                     36221b12-f362-4dca-8724-4da82ef2ba49: "ROLE_ADMIN" # Sarah Khalil
+                    %sensiolabs_connect.app_id%: "ROLE_ADMIN"
```
Comment (jderusse) [inline app/config/security.yml]: A UserId is expected here

File: bin/doctrine-migrations [hunk 3]
```diff
@@ -0,0 +1 @@
+../vendor/doctrine/migrations/bin/doctrine-migrations
```
Comment (jderusse) [inline bin/doctrine-migrations]: You should not commit generated files. 
Add this file in the `.gitignore`

File: app/config/routing.yml [hunk 4]
```diff
@@ -5,3 +5,6 @@ sensio_labs_job_board:
 
 _sensiolabs_connect:
     resource: "@SensioLabsConnectBundle/Resources/config/routing.xml"
+
+oauth_logout:
+    pattern: /session/logout
```
Comment (mmarchois) [inline app/config/routing.yml]: Missing blank line at end of file

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    /**
+     * @Conf\Route("/session/callback", name="oauth_callback")
+     */
+    public function newSessionAction(Request $request)
+    {
+        return $this->get('security.authentication.entry_point.sensiolabs_connect')->start($request);
+    }
+}
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: Missing blank line at end of file

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller as BaseController;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration as Conf;
+use SensioLabs\Connect\Security\Authentication\Token\ConnectToken;
+
+class SlnController extends BaseController
+{
+    /**
+     * @Conf\Route("/sln_customiser.js", name="sln_customize")
```
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: `@Route` ?

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace SensioLabs\JobBoardBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller as BaseController;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration as Conf;
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: Usely, we import the Class `Route` to directly use the annotation `@Route`

File: src/SensioLabs/JobBoardBundle/Resources/views/Sln/index.js.twig [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    '0': { 'url': '{{ path('oauth_logout') }}', 'label' : "Logout" }
+    };
+{% else %}
+    SLNBar.isAuthenticated = false;
+    SLNBar.urlConnectButton = '{{ path('sensiolabs_connect_new_session') }}';
+{% endif %}
+
+$('#sln').html(SLNBar.render());
+SLNBar.bindEvents();
+});
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Resources/views/Sln/index.js.twig]: Missing blank line at end of file

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+use SensioLabs\Connect\Security\Authentication\Token\ConnectToken;
+
+class SlnController extends BaseController
+{
+    /**
+     * @Conf\Route("/sln_customiser.js", name="sln_customize")
+     * @Conf\Template("SensiolabsJobBoardBundle:Sln:index.js.twig")
```
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: `@Template` ?

File: .gitignore (hunk 1 above)
Comment (romqin) [inline .gitignore]: It could be in your [global .gitignore](https://help.github.com/articles/ignoring-files/).

Comment (ogizanagi) [inline .gitignore]: it SHOULD be :smile:

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    public function indexAction()
+    {
+        $token = $this->get('security.context')->getToken();
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: `security.token_storage` ?

File: app/config/routing.yml (hunk 4 above)
Comment (jderusse) [inline app/config/routing.yml]: Url don't match the one you defined in `security.yml`

File: app/config/security.yml [hunk 11]
```diff
@@ -20,7 +21,9 @@ security:
                 login_path:  /login
                 remember_me: false
                 provider:    sensiolabs_connect
-            logout:          ~
+            logout:
+              path:   /logout
```
Comment (salahm) [inline app/config/security.yml]: i'm not sure, this is really needed. it's default values.

File: src/SensioLabs/JobBoardBundle/Resources/views/layout.html.twig [hunk 12]
```diff
@@ -81,5 +81,27 @@
             <!--[if lt IE 9]><script src="{{ asset('bundles/sensiolabsjobboard/js/html5.js') }}"></script><![endif]-->
             {{ include('SensioLabsJobBoardBundle:Connect:include.html.twig') }}
         {% endblock %}
+        
+        <script>
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/views/layout.html.twig]: should be in a separated js file.

File: src/SensioLabs/JobBoardBundle/Controller/SlnController.php (hunk 9 above)
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: this js file doesn't exists.

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Controller/SlnController.php]: I don't think it's a good idea to use the `@Template` (in general), but especially for generating js files. How is set the `Content-Type` of the response for instance ?

File: .gitignore [hunk 13]
```diff
@@ -4,6 +4,7 @@ app/config/parameters.yml
 app/logs/*
 app/sessions/*
 app/Resources/dump
+bin/doctrine-migrations
```
Comment (jderusse) [inline .gitignore]: Don't forget to remove the file to `git rm`

**Comment by mRoca** [comment]
Now, you can squash your commits : `git rebase -i HEAD~5`

File: src/SensioLabs/JobBoardBundle/Resources/views/layout.html.twig [hunk 14]
```diff
@@ -81,5 +81,6 @@
             <!--[if lt IE 9]><script src="{{ asset('bundles/sensiolabsjobboard/js/html5.js') }}"></script><![endif]-->
             {{ include('SensioLabsJobBoardBundle:Connect:include.html.twig') }}
         {% endblock %}
+
```
Comment (romqin) [inline src/SensioLabs/JobBoardBundle/Resources/views/layout.html.twig]: You should (or maybe must) use `git add -p` to avoid to add useless modifications.

**Pull request merged**